core/src/main/java/nl/nn/adapterframework/jta/narayana/NarayanaTransactionHelper.java

```diff
@@ -0,0 +1,63 @@
+/*
+   Copyright 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.jta.narayana;
+
+import javax.jms.JMSException;
+import javax.transaction.TransactionManager;
+
+import org.jboss.narayana.jta.jms.ConnectionProxy;
+import org.jboss.narayana.jta.jms.SessionProxy;
+import org.jboss.narayana.jta.jms.TransactionHelperImpl;
+
+import com.arjuna.ats.arjuna.coordinator.ActionStatus;
+import com.arjuna.ats.internal.arjuna.coordinator.ReaperThread;
+import com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple;
+import com.arjuna.ats.internal.jta.utils.arjunacore.StatusConverter;
+
+public class NarayanaTransactionHelper extends TransactionHelperImpl {
+
+	private final TransactionManager transactionManager;
+
+	public NarayanaTransactionHelper(TransactionManager transactionManager) {
+		super(transactionManager);
+		this.transactionManager = transactionManager;
+	}
+
+	/**
+	 * Connections are not always closed. {@link ConnectionProxy#close() } and {@link SessionProxy#close() } 
```

Comment:
```suggestion
	 * Connections were not always closed, because the super implementation of this method returns false too often. {@link ConnectionProxy#close() } and {@link SessionProxy#close() } 
```